Do not  throw if Git is upgraded  while GE is started

gerhardol opened the pull request:
Related to the closed issue of #6692 and others
Submitted to minimize time wasted on this again.

With regards to https://github.com/gitextensions/gitextensions/issues/6692#issuecomment-710881897
This change is like allowing the user to hot swap the engine while running, as long as they do not touch any controls,
or they have changed specific addons.

## Proposed changes

Ignore exceptions for GitStatusMonitor and Background fetch plugin,
as well as "Browse activation" (selecting the window).
This is the most common background use of Git.
The primary usage is to allow Git upgrades while GE is active.

Active use of GE will raise exceptions, not much can be done when Git is not available.
#8278 or #8358 improves the error message.

Note that GitStatusMonitor will not monitor changes until refresh or a new repo is opened,
and that possible merge conflicts will be ignored (to not trigger the user to select Resolve).

## Test methodology <!-- How did you ensure quality? -->

Manual, removing git.exe after startup

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

RussKie commented:
![image](https://user-images.githubusercontent.com/4403806/97801372-62007e80-1c90-11eb-9ef1-a161bf979cba.png)

Did you run of of things to say? :)

gerhardol commented:
> Did you run of of things to say? :)

I added this to the top instead, changed.

I do not think this is controversial, but will give it a few days before merging for comments.

gerhardol commented:
@msftbot merge in 3 days
